strawberryfields/engine.py

```diff
@@ -531,161 +544,541 @@ def run(self, program, *, args=None, compile_options=None, run_options=None):
+            str: the base URL
         """
-        Returns a Blackbird object to be sent later to the server when creating a job.
-        Assumes the current backend as the target.
+        return "http{}://{}:{}".format(
+            "s" if self.use_ssl else "", self.host, self.port
+        )
+
+    def create_job(self, target: str, program: Program, shots: int) -> Job:
+        """Creates a job with the given circuit.
 
         Args:
+            target (str): the target device
+            program (strawberryfields.Program): the quantum circuit
             shots (int): the number of shots
-            program (Program): program to be converted to Blackbird code
 
         Returns:
-            blackbird.BlackbirdProgram
+            strawberryfields.engine.Job: the created job
         """
-        bb = to_blackbird(program, version="1.0")
+        # Serialize a blackbird circuit for network transmission
+        bb = to_blackbird(program)
+        # pylint: disable=protected-access
+        bb._target["name"] = target
+        # pylint: disable=protected-access
+        bb._target["options"] = {"shots": shots}
+        circuit = bb.serialize()
+
+        response = self._post("/jobs", data=json.dumps({"circuit": circuit}))
+        if response.status_code == 201:
+            return Job(
+                id_=response.json()["id"],
+                status=JobStatus(response.json()["status"]),
+                connection=self,
+            )
+        raise RequestFailedError(
+            "Job creation failed: {}".format(self._format_error_message(response))
+        )
+
+    def get_all_jobs(self, after: datetime = datetime(1970, 1, 1)) -> List[Job]:
+        """Gets a list of jobs created by the user, optionally filtered by datetime.
 
-        # TODO: This is potentially not needed here
-        bb._target["name"] = self.backend_name
-        bb._target["options"] = {"shots": shots, **program.backend_options}
-        return bb
+        A maximum of the 100 most recent jobs are returned.
+
+        Args:
+            after (datetime.datetime): if provided, only jobs more recently created
+                then ``after`` are returned
 
-    def _queue_job(self, job_content):
+        Returns:
+            List[strawberryfields.engine.Job]: the jobs
         """
-        Create a Job instance based on job_content, and send the job to the API. Append to list
-        of jobs.
+        # response = self._get("/jobs?size={}".format(self.MAX_JOBS_REQUESTED))
+        # if response.status_code == 200:
+            # return [
+                # Job(id_=info["id"], status=info["status"], connection=self)
+                # for info in response.json()["data"]
+                # if datetime.strptime(info["created_at"], self.JOB_TIMESTAMP_FORMAT)
+                # > after
+            # ]
+        # raise RequestFailedError(self._format_error_message(response))
+        raise NotImplementedError("This feature is not yet implemented")
```

Comment:
I've left the above commented for context, but I think we probably want to remove it before merging.

Reply:
On further thought, I wonder if it's worth having this method stub here at all, or if we should just remove it and put it back in a future PR. Thoughts @antalszava @josh146?